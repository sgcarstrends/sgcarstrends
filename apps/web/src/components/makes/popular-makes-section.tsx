import { Chip } from "@heroui/chip";
import type { Make } from "@web/types";
import { MakeCard } from "./make-card";

interface PopularMakesSectionProps {
  makes: Make[];
  onMakePress?: (make: Make) => void;
}

export const PopularMakesSection = ({
  makes,
  onMakePress,
}: PopularMakesSectionProps) => {
  if (makes.length === 0) {
    return null;
  }

  return (
    <section className="space-y-4">
      <div className="flex items-center gap-2">
        <h2 className="font-semibold text-xl">Popular Makes</h2>
        <Chip size="sm" variant="shadow" color="primary">
          {makes.length}
        </Chip>
      </div>
      <div className="grid grid-cols-2 gap-4 md:grid-cols-4">
        {makes.map((make) => (
          <MakeCard
            key={make}
            make={make}
            isPopular={true}
            onMakePress={onMakePress}
          />
        ))}
      </div>
    </section>
  );
};
