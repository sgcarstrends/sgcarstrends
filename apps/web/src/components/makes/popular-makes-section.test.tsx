import { render, screen } from "@testing-library/react";
import { PopularMakesSection } from "./popular-makes-section";

describe("PopularMakesSection", () => {
  const germanMakes = ["AUDI", "BMW", "MERCEDES BENZ", "VOLKSWAGEN"];

  it("should render the section heading and count", () => {
    render(<PopularMakesSection makes={germanMakes} />);
    expect(screen.getByText("Popular Makes")).toBeVisible();
    expect(screen.getByText(germanMakes.length.toString())).toBeVisible();
  });

  it("should render a card for each popular make", () => {
    render(<PopularMakesSection makes={germanMakes} />);
    germanMakes.forEach((make) => {
      expect(screen.getByText(make)).toBeVisible();
    });
  });

  it("should highlight each card as popular", () => {
    render(<PopularMakesSection makes={germanMakes} />);
    expect(screen.getAllByText("Popular")).toHaveLength(germanMakes.length);
  });

  it("should render nothing when no popular makes exist", () => {
    const { container } = render(<PopularMakesSection makes={[]} />);
    expect(container).toBeEmptyDOMElement();
  });
});
