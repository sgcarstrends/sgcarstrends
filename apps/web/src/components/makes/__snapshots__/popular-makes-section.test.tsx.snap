// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PopularMakesSection > should render correctly 1`] = `
<div>
  <section
    class="space-y-4"
  >
    <div
      class="flex items-center gap-2"
    >
      <h2
        class="font-semibold text-xl"
      >
        Popular Makes
      </h2>
      <div
        class="relative max-w-fit min-w-min inline-flex items-center justify-between box-border whitespace-nowrap px-1 h-6 text-tiny rounded-full shadow-lg shadow-primary/40 bg-primary text-primary-foreground"
      >
        <span
          class="flex-1 text-inherit font-normal px-1"
        >
          3
        </span>
      </div>
    </div>
    <div
      class="grid grid-cols-2 gap-4 md:grid-cols-4"
    >
      <div
        class="flex flex-col relative overflow-hidden text-foreground box-border bg-content1 outline-hidden data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-medium rounded-large transition-transform-background motion-reduce:transition-none h-32 border-none ring-2 ring-primary-600"
        tabindex="-1"
      >
        <div
          class="flex h-full w-full justify-center"
        >
          <img
            alt="TOYOTA Logo"
            class="object-contain"
            data-nimg="1"
            decoding="async"
            height="512"
            loading="lazy"
            src="/_next/image?url=https%3A%2F%2Fassets.sgcarstrends.com%2Flogos%2Ftoyota.png&w=1080&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fassets.sgcarstrends.com%2Flogos%2Ftoyota.png&w=640&q=75 1x, /_next/image?url=https%3A%2F%2Fassets.sgcarstrends.com%2Flogos%2Ftoyota.png&w=1080&q=75 2x"
            style="color: transparent;"
            width="512"
          />
          <div
            class="fallback-icon hidden h-20 w-20 items-center justify-center rounded-lg bg-primary-100"
          >
            <svg
              class="lucide lucide-car size-8 text-primary-600"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19 17h2c.6 0 1-.4 1-1v-3c0-.9-.7-1.7-1.5-1.9C18.7 10.6 16 10 16 10s-1.3-1.4-2.2-2.3c-.5-.4-1.1-.7-1.8-.7H5c-.6 0-1.1.4-1.4.9l-1.4 2.9A3.7 3.7 0 0 0 2 12v4c0 .6.4 1 1 1h2"
              />
              <circle
                cx="7"
                cy="17"
                r="2"
              />
              <path
                d="M9 17h6"
              />
              <circle
                cx="17"
                cy="17"
                r="2"
              />
            </svg>
          </div>
        </div>
        <div
          class="p-3 h-auto flex items-center color-inherit subpixel-antialiased bg-background/10 backdrop-saturate-150 absolute bottom-1 z-10 ml-1 w-[calc(100%_-_8px)] justify-between overflow-hidden rounded-large border-1 border-white/20 py-1 shadow-small backdrop-blur before:rounded-xl before:bg-black/50"
        >
          <p
            class="text-tiny"
          >
            TOYOTA
          </p>
          <a
            class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] cursor-pointer outline-hidden data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-3 min-w-16 h-8 text-tiny gap-2 rounded-large [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none shadow-lg shadow-primary/40 bg-primary text-primary-foreground data-[hover=true]:opacity-hover"
            data-react-aria-pressable="true"
            href="/cars/makes/toyota"
            role="button"
            tabindex="0"
          >
            Explore
          </a>
        </div>
      </div>
      <div
        class="flex flex-col relative overflow-hidden text-foreground box-border bg-content1 outline-hidden data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-medium rounded-large transition-transform-background motion-reduce:transition-none h-32 border-none ring-2 ring-primary-600"
        tabindex="-1"
      >
        <div
          class="flex h-full w-full justify-center"
        >
          <img
            alt="HONDA Logo"
            class="object-contain"
            data-nimg="1"
            decoding="async"
            height="512"
            loading="lazy"
            src="/_next/image?url=https%3A%2F%2Fassets.sgcarstrends.com%2Flogos%2Fhonda.png&w=1080&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fassets.sgcarstrends.com%2Flogos%2Fhonda.png&w=640&q=75 1x, /_next/image?url=https%3A%2F%2Fassets.sgcarstrends.com%2Flogos%2Fhonda.png&w=1080&q=75 2x"
            style="color: transparent;"
            width="512"
          />
          <div
            class="fallback-icon hidden h-20 w-20 items-center justify-center rounded-lg bg-primary-100"
          >
            <svg
              class="lucide lucide-car size-8 text-primary-600"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19 17h2c.6 0 1-.4 1-1v-3c0-.9-.7-1.7-1.5-1.9C18.7 10.6 16 10 16 10s-1.3-1.4-2.2-2.3c-.5-.4-1.1-.7-1.8-.7H5c-.6 0-1.1.4-1.4.9l-1.4 2.9A3.7 3.7 0 0 0 2 12v4c0 .6.4 1 1 1h2"
              />
              <circle
                cx="7"
                cy="17"
                r="2"
              />
              <path
                d="M9 17h6"
              />
              <circle
                cx="17"
                cy="17"
                r="2"
              />
            </svg>
          </div>
        </div>
        <div
          class="p-3 h-auto flex items-center color-inherit subpixel-antialiased bg-background/10 backdrop-saturate-150 absolute bottom-1 z-10 ml-1 w-[calc(100%_-_8px)] justify-between overflow-hidden rounded-large border-1 border-white/20 py-1 shadow-small backdrop-blur before:rounded-xl before:bg-black/50"
        >
          <p
            class="text-tiny"
          >
            HONDA
          </p>
          <a
            class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] cursor-pointer outline-hidden data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-3 min-w-16 h-8 text-tiny gap-2 rounded-large [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none shadow-lg shadow-primary/40 bg-primary text-primary-foreground data-[hover=true]:opacity-hover"
            data-react-aria-pressable="true"
            href="/cars/makes/honda"
            role="button"
            tabindex="0"
          >
            Explore
          </a>
        </div>
      </div>
      <div
        class="flex flex-col relative overflow-hidden text-foreground box-border bg-content1 outline-hidden data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-medium rounded-large transition-transform-background motion-reduce:transition-none h-32 border-none ring-2 ring-primary-600"
        tabindex="-1"
      >
        <div
          class="flex h-full w-full justify-center"
        >
          <img
            alt="BMW Logo"
            class="object-contain"
            data-nimg="1"
            decoding="async"
            height="512"
            loading="lazy"
            src="/_next/image?url=https%3A%2F%2Fassets.sgcarstrends.com%2Flogos%2Fbmw.png&w=1080&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fassets.sgcarstrends.com%2Flogos%2Fbmw.png&w=640&q=75 1x, /_next/image?url=https%3A%2F%2Fassets.sgcarstrends.com%2Flogos%2Fbmw.png&w=1080&q=75 2x"
            style="color: transparent;"
            width="512"
          />
          <div
            class="fallback-icon hidden h-20 w-20 items-center justify-center rounded-lg bg-primary-100"
          >
            <svg
              class="lucide lucide-car size-8 text-primary-600"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19 17h2c.6 0 1-.4 1-1v-3c0-.9-.7-1.7-1.5-1.9C18.7 10.6 16 10 16 10s-1.3-1.4-2.2-2.3c-.5-.4-1.1-.7-1.8-.7H5c-.6 0-1.1.4-1.4.9l-1.4 2.9A3.7 3.7 0 0 0 2 12v4c0 .6.4 1 1 1h2"
              />
              <circle
                cx="7"
                cy="17"
                r="2"
              />
              <path
                d="M9 17h6"
              />
              <circle
                cx="17"
                cy="17"
                r="2"
              />
            </svg>
          </div>
        </div>
        <div
          class="p-3 h-auto flex items-center color-inherit subpixel-antialiased bg-background/10 backdrop-saturate-150 absolute bottom-1 z-10 ml-1 w-[calc(100%_-_8px)] justify-between overflow-hidden rounded-large border-1 border-white/20 py-1 shadow-small backdrop-blur before:rounded-xl before:bg-black/50"
        >
          <p
            class="text-tiny"
          >
            BMW
          </p>
          <a
            class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] cursor-pointer outline-hidden data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-3 min-w-16 h-8 text-tiny gap-2 rounded-large [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none shadow-lg shadow-primary/40 bg-primary text-primary-foreground data-[hover=true]:opacity-hover"
            data-react-aria-pressable="true"
            href="/cars/makes/bmw"
            role="button"
            tabindex="0"
          >
            Explore
          </a>
        </div>
      </div>
    </div>
  </section>
</div>
`;
